src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -37,7 +37,7 @@ public function validate($value, Constraint $constraint)
         $valid = filter_var($value, FILTER_VALIDATE_EMAIL);
 
         if ($valid) {
-            $host = substr($value, strpos($value, '@') + 1);
+            $host = $this->extractHostname($value);
```

Comment:
This code was extracted into its own method only so we can unit test it in isolation.